tests/unit/staff/views/test_applications.py

```diff
@@ -137,18 +137,20 @@ def test_applicationlist_search(rf, core_developer):
     assert set_from_qs(response.context_data["object_list"]) == {app1.pk, app2.pk}
 
 
-def test_applicationdetail_submitted_review_updates_last_reviewed(
+def test_applicationdetail_updates_review_details_following_review(
```

Comment:
Rather than adding a new test here, I'd be tempted to include these tests in one or both of `test_applicationdetail_post_with_complete_application` / `test_applicationdetail_post_with_incomplete_application`.